### PR TITLE
Add connect on matrix

### PR DESCRIPTION
**Pull request by MV-GH**
Add connect on matrix feature, this will open the relevant matrix client with the option to dm this user.

I designed the SVG icon of Matrix myself. So hereby I release the icon into the public domain

matrix_icon

Edit:
Someone linked me a smaller icon, I am using that one now.

image

### CODE REVIEW

File: app/src/main/java/com/jerboa/ui/components/common/DropdownMenu.kt [hunk 1]
```diff
@@ -40,6 +40,29 @@ fun MenuItem(
...
 }
 
+@Composable
+fun MenuItem(
+    text: String,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    textModifier: Modifier = Modifier,
+    icon: @Composable (() -> Unit),
+    textStyle: TextStyle = MaterialTheme.typography.bodyMedium,
```
Comment (dessalines) [inline app/src/main/java/com/jerboa/ui/components/common/DropdownMenu.kt:50]: Ah I see, mainly for the icon.

File: app/src/main/java/com/jerboa/ui/components/person/PersonProfile.kt [hunk 2]
```diff
@@ -278,6 +289,7 @@ fun PersonProfileMoreDropdown(
     onBlockPersonClick: () -> Unit,
     onReportPersonClick: () -> Unit,
     onMessagePersonClick: () -> Unit,
+    openMatrix: (() -> Unit)?,
```
Comment (dessalines) [inline app/src/main/java/com/jerboa/ui/components/person/PersonProfile.kt:292]: Hrm.... not a fan of optional params, as they often result in places you don't catch them. Probably best would be 

```
hasMatrix: Boolean,
onOpenMatrixClick: () -> Unit
```

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -412,4 +412,5 @@
     <string name="open_link_external">Open link in external</string>
     <string name="no_activity_found">No activity (app) found that can open this link</string>
     <string name="markdownHelper_insertSpoiler">Insert spoiler</string>
+    <string name="matrix_send_msg">Connect on Matrix</string>
```
Comment (dessalines) [inline app/src/main/res/values/strings.xml:415]: I don't know that I've seen this phrase used before, but it makes sense, and is concise.

File: app/src/main/java/com/jerboa/ui/components/person/PersonProfile.kt (hunk 2 above)
Comment (MV-GH) [reply app/src/main/java/com/jerboa/ui/components/person/PersonProfile.kt:292]: Optional params have indeed been a footgun but this param is not optional. You are forced to pass null or a lambda. Not doing will result in a compile error. *That would not be the case if I did `openMatrix: (() -> Unit)? = null`*

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (MV-GH) [reply app/src/main/res/values/strings.xml:415]: Yeah, I prefer it to `Send secure message through Matrix` or `Send a secure message`.

File: app/src/main/java/com/jerboa/ui/components/person/PersonProfile.kt (hunk 2 above)
Comment (dessalines) [reply app/src/main/java/com/jerboa/ui/components/person/PersonProfile.kt:292]: mmmk cool.

**Pull request merged**